### PR TITLE
Add Travis CI support

### PR DESCRIPTION
**Pull request by yjpa7145**

### CODE REVIEW

File: travis-ci/deploy-to-npm.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
+VERSION=$(jq --raw-output .version lerna.json)
+./node_modules/.bin/lerna publish \
+  --skip-git \
+  --repo-version "$VERSION" \
+  --yes \
+  --force-publish=* \
+  --npm-client=npm \
+  --npm-tag="mth-test-$VERSION"
```
Comment (laurenfrederick) [inline travis-ci/deploy-to-npm.sh]: You probably want to remove `--npm-tag="mth-test-$VERSION"`

Comment (laurenfrederick) [reply travis-ci/deploy-to-npm.sh]: For npm though we need to be able to use a tag other than latest. I was doing that based on the branch name before.

File: README.md [hunk 2]
```diff
@@ -27,13 +26,6 @@ nvm install
 nvm use
 ```
 
-Ensure that the aws cli is configured and that the default output format is either JSON or None:
```
Comment (yjpa7145) [inline README.md]: If we are using the AWS CLI in a script somewhere and care about the format of the output, we should be using the `--json` flag in that script.

File: .nvmrc [hunk 3]
```diff
@@ -1 +1 @@
-8.11
+8.10
```
Comment (yjpa7145) [inline .nvmrc]: Lambda runs 8.10, not 8.11.

File: package.json [hunk 4]
```diff
@@ -10,9 +10,9 @@
     "e2e": "ava tests/*.js --serial",
     "lint": "eslint packages/api --ext .js",
     "test": "nyc lerna run test --concurrency 2",
-    "bootstrap": "lerna bootstrap --npm-client=npm",
+    "bootstrap": "lerna bootstrap --npm-client=npm --hoist",
```
Comment (yjpa7145) [inline package.json]: Adding `--hoist` drastically cut down on the size of the build cache.

File: packages/api/package.json [hunk 5]
```diff
@@ -37,7 +37,7 @@
     "@cumulus/cmrjs": "^1.9.0",
     "@cumulus/common": "^1.9.0",
     "@cumulus/ingest": "^1.9.0",
-    "@cumulus/pvl": "^1.8.0",
+    "@cumulus/pvl": "^1.9.0",
```
Comment (yjpa7145) [inline packages/api/package.json]: Not sure why this was 1.8.0 when all of the other versions were 1.9.0.

File: example/README.md [hunk 6]
```diff
@@ -71,15 +71,10 @@ The workflow should be configured as it would be for a normal Cumulus deployment
...
-To use your own CF stack for running integration tests in CircleCI builds, add your github username and your kes deployment name [here](spec/select#L5).
-
-Example:
+## Using your AWS CF stack in Travis CI
 
-```bash
-   developers=( ["myexample_github_username"]="mykesdeploymentname" )
-```
+To use your own CF stack for running integration tests in Travis CI builds, add
+your stack name [here](../travis-ci/selec-stack.js).
```
Comment (ifestus) [inline example/README.md]: should probably be `../travis-ci/select-stack.js`

File: packages/common/test/config/test-collections.yml [hunk 7]
```diff
@@ -9,7 +9,7 @@ providers:
         activity: SIPSPROVDownloadActivity
         conn_config:
           conn_type: ftp
-          host: localhost
+          host: 127.0.0.1
```
Comment (yjpa7145) [inline packages/common/test/config/test-collections.yml]: In Travis CI `localhost` can resolve to an IPv6 address, but the docker containers are only listening on the IPv4 address.

File: packages/deployment/lib/adapter.js [hunk 8]
```diff
@@ -65,8 +65,12 @@ function extractZipFile(filename, dst) {
  * @returns {Promise.<string>} Promise resolution is string of latest github release, e.g. 'v0.0.1'
  */
 function fetchLatestMessageAdapterRelease(gitPath) {
+  const url = process.env.GITHUB_TOKEN ?
```
Comment (yjpa7145) [inline packages/deployment/lib/adapter.js]: By authenticating with the `access_token` we prevent throttling issues.

File: README.md [hunk 9]
```diff
@@ -82,26 +74,58 @@ Run the test commands next
...
+
+In order to prevent multiple instances of the integration tests from running
+against a stack at the same time, a lock file is created in S3 for each stack.
+Before integration tests start they will wait until that lock file is not
+present. They will then create that lock file, run the tests, and delete the
+lock file. The lock file will be located at
+`s3://${CACHE_BUCKET}/travis-ci-integration-tests/${DEPLOYMENT}.lock`. The lock
+file will contain a link to the Travis CI job that created the lock file. If
+your tests seem to be hung waiting for that lock file, check to see if the job
+that created the lock file is still runnging or has crashed. If it has crashed
```
Comment (laurenfrederick) [inline README.md]: running

File: README.md [hunk 10]
```diff
@@ -82,26 +74,58 @@ Run the test commands next
...
+then the lock file should be deleted. You should also figure out why the lock
+file was not cleaned up and fix that for next time.
 
 ### Code Coverage
 
-Code coverage is checked using [nyc](https://github.com/istanbuljs/nyc). The CircleCI build tests coverage. A summary can be viewed in the build output. Detailed code coverage in html can be found by going to the Artifacts tab and navigating to `index.html` in the coverage folder. Clicking on `index.html` will take you to an html page showing code coverage for each individual file.
+Code coverage is checked using [nyc](https://github.com/istanbuljs/nyc). The
+Travis CI build tests coverage. A summary can be viewed in the build output.
```
Comment (laurenfrederick) [inline README.md]: builds

File: README.md [hunk 11]
```diff
@@ -164,99 +188,71 @@ Read more about the semantic versioning [here](https://docs.npmjs.com/getting-st
...
-Lerna handles the process of deciding which version number should be used as long as the developer decides whether the change is a patch or a minor/major change.
+Lerna handles the process of deciding which version number should be used as
+long as the developer specifies whether the change is a major, minor, or patch
+change.
 
 To update cumulus' version number run:
 
      $ yarn update
 
-You will be prompted to select the type of change (patch/minor/major). Lerna will update the version of all packages after the selection.
```
Comment (laurenfrederick) [inline README.md]: Why did you remove this?

File: README.md [hunk 12]
```diff
@@ -164,99 +188,71 @@ Read more about the semantic versioning [here](https://docs.npmjs.com/getting-st
...
 
-Push the tag to github
+Create a PR against the `master` branch. Verify that the Travis CI build for the
+PR succeeds.  Once merged, the release branch can be deleted.
 
-  $ git push origin v1.x.x
+#### 6. Create a git tag for the release
 
-#### 6. PR and merge to master
+Publishing of new releases is handled by Travis CI and is triggered by the
```
Comment (laurenfrederick) [inline README.md]: Where in that process should this be done? Right after merge to master? I think we should specify that pushing the tag actually triggers the release build.

File: README.md (hunk 9 above)
Comment (yjpa7145) [reply README.md]: Fixed.

File: README.md (hunk 10 above)
Comment (yjpa7145) [reply README.md]: Changed.

File: README.md (hunk 11 above)
Comment (yjpa7145) [reply README.md]: It's redundant between the comment about "Lerna handles the process of deciding which version number should be used as long as the developer specifies whether the change is a major, minor, or patch change" and the screenshot.

File: README.md (hunk 12 above)
Comment (yjpa7145) [reply README.md]: Yes, it should be done after the merge to master.  I've updated steps 5 and 6 to make it more explicit.
